### ISSUE TITLE
Refilling water from the furthest possible source?

### ISSUE THREAD

**Issue by tukoz** [Dubwise56/Dubs-Bad-Hygiene#45]
Hi, so pawn looks for the "fastest" water source (whatever time it'll take to go and return), bypassing any check for the closest source:
https://www.casimages.com/i/20123104411518649217192990.jpg.html
Unless we manually forbid all the "fastest" water sources available on the map:
https://www.casimages.com/i/20123104411418649217192988.jpg.html
https://www.casimages.com/i/20123104411418649217192989.jpg.html
, right? 
Any way around this beside going FULL plumbing when we don't want that?

**Comment by Dubwise56** [comment]
its based on the fixture quality, else pawns would be using puddles of water or water tubs when there's nice high quality power showers just 3 cells further away, they will only fall back to using the well in that situation if your plumbing system stops working or becomes contaminated, so you just need to replace your water tub with a sink or fountain once you upgrade to plumbing, or replace the well with a sink

**Comment by tukoz** [comment]
So I'm forced to upgrade, hmm (I honestly think a water well should be part of the good/top/primary sources of water). Does what you kindly detailed also explain why pawns will ignore the wash tube next to them and cross half the base (and what many doors) to wash their hands in a power-shower? 
Your very quick answer's much appreciated here Dubwise!

**Comment by Dubwise56** [comment]
well for using fixtures its much more complicated there's range limits involved based on need levels and what not, and you're not forced to upgrade, you can just let your pawn fill the tub up from that tap, its only 20 odd cells away once every x days when the tub actually run out of uses, thats nothing compared to the time they spend pathing hundreds of cells just to sweep up some dirt or repair something that breaks down

**Comment by tukoz** [comment]
That's a comforting fact for the New Year, thx Dubwise! And.. would it would be impolite to ask for a little change so that pawns don't prefer to wash their hands at the power shower half the base away over the shiny marble **basin** next to the toilet?

**Comment by Dubwise56** [comment]
that shouldn't be possible, the scan for washing hands after using a toilet has a very limited range

**Comment by tukoz** [comment]
Then Steam lent me a version with the opposite way of scanning i.e. start at max range when washing hands:
https://www.casimages.com/i/21010108000618649217194308.jpg.html That one is not the farthest I've seen by far, but both basins are the exact same, both are free and pawn goes for the furthest. He's an undergrounder btw.

**Comment by Dubwise56** [comment]
whaaa, ok that shouldn't be possible, i'll check it

**Comment by Dubwise56** [comment]
ok yup the search range for washing hands is 30 cells which is way too high for hand washing, it was supposed to be more like 10, must have got missed during a refactor

**Comment by Dubwise56** [comment]
yup i found the old code and it was a 10 cell limit which got replaced with a 30 cell limit during a refactor, will be fixed for next update, they will still want to use the highest quality fixture to wash their hands with though, so if there's a bucket next to the shitter and a kitchen sink in another room within 10 cells they will use the kitchen sink, which i personally think is realistic and so makes more sense

**Comment by tukoz** [comment]
Great that you found it @Dubwise56. Also you found the hiccup the very day I passed some related info to you. I made a bunch of screens for days after without checking this Github page. Maybe I should check my own "code" hmmm

> If there's a bucket next to the shitter and a kitchen sink in another room within 10 cells they will use the kitchen sink, which i personally think is realistic and so makes more sense.

For tribals not sure really. But the 10 cells radius should make for it. Can't wait for the next (Steam) update!

**Comment by tukoz** [comment]
@Dubwise56 I was wondering if you could share some news about the update? The « Use furthest hygiene furniture available » is... painful, especially when they're bleeding or crawling at 1.0 speed for serious health reasons:  
https://www.casimages.com/i/21021302262618649217259718.jpg.html  
https://www.casimages.com/i/21021302262618649217259719.jpg.html  
https://www.casimages.com/i/21021302262618649217259720.jpg.html  
https://www.casimages.com/i/21021302262618649217259721.jpg.html  
https://www.casimages.com/i/21021302262718649217259722.jpg.html

**Comment by Dubwise56** [comment]
i already pushed that fix to the master back when i found the issue

**Comment by tukoz** [comment]
Oh. Are the ranges in the linked screenshots those intended? They're taken in a new game, with the latest available Steam versions of both game and mods (also after removing all files then reinstall).

**Comment by Dubwise56** [comment]
its not in the steam version yet just the master on git

**Comment by tukoz** [comment]
OK I'll fetch it thanx. Would you mind to share with us about the date of the upcoming update of the Steam version of your mod please?
